drivers/volume/portworx/portworx.go

```diff
@@ -3238,17 +3238,41 @@ func (d *portworx) DecommissionNode(n *node.Node) error {
 	}
 	_, err = task.DoRetryWithTimeout(t, defaultTimeout, defaultRetryInterval)
 
-	// TODO replace when sdk supports node removal
-	if err = d.legacyClusterManager.Remove([]api.Node{{Id: nodeResp.Node.Id}}, true); err != nil {
-		if !strings.Contains(err.Error(), "Node remove is pending") {
-			return &ErrFailedToDecommissionNode{
-				Node:  n.Name,
-				Cause: err.Error(),
-			}
+	cmd := fmt.Sprintf("echo Y | %s cluster delete -f %s", d.getPxctlPath(*n), nodeResp.Node.Id)
+	log.Infof("Running command [%s] on node [%s]", cmd, n.Name)
+
+	var cmdNode node.Node
+	for _, cn := range node.GetStorageDriverNodes() {
+		if cn.Name != n.Name {
+			cmdNode = cn
+			break
+		}
+	}
+
+	t = func() (interface{}, bool, error) {
+		out, err := d.nodeDriver.RunCommandWithNoRetry(cmdNode, cmd, node.ConnectionOpts{
+			Timeout:         2 * time.Minute,
+			TimeBeforeRetry: 10 * time.Second,
+		})
+		if err != nil && strings.Contains(err.Error(), "Node remove is pending") {
+
+			return out, false, nil
+
 		}
+		if err != nil {
+			return "", true, err
```

Comment:
Why do you want to retry in case of error? Here the error would mean the command did not exit with status 0 right?